qa/tasks/cephfs/test_volumes.py

```diff
@@ -92,8 +96,28 @@ def _verify_clone_attrs(self, subvolume, clone, source_group=None, clone_group=N
             cval = int(self.mount_a.run_shell(['stat', '-c' '%Y', sink_path]).stdout.getvalue().strip())
             self.assertEqual(sval, cval)
 
-    def _verify_clone(self, subvolume, clone, source_group=None, clone_group=None, timo=120):
-        path1 = self._get_subvolume_path(self.volname, subvolume, group_name=source_group)
+    def _verify_clone_root(self, subvolume, snapshot, clone, source_group, clone_group, clone_pool):
+        clone_info = json.loads(self._get_subvolume_info(self.volname, clone, clone_group))
+        # verify size is inherited from source subvolume
+        # TODO: enable once quota source is determined and set on the clone
+        # source_info = json.loads(self._get_subvolume_info(self.volname, subvolume, source_group))
+        # self.assertEqual(clone_info["bytes_quota"], source_info["bytes_quota"])
+
+        if clone_pool:
+            # verify pool is set as per request
+            clone_path = self._get_subvolume_path(self.volname, clone)
+            ondisk_pool = self.mount_a.getfattr(clone_path, "ceph.dir.layout.pool")
+            self.assertEqual(ondisk_pool, clone_pool)
+        else:
+            # verify pool and pool namespace are inherited from snapshot
+            source_path = self._get_subvolume_snapshot_path(subvolume, snapshot, source_group)
+            self.assertEqual(clone_info["data_pool"],
+                             self.mount_a.getfattr(source_path, "ceph.dir.layout.pool"))
+            self.assertEqual(clone_info["pool_namespace"],
```

Comment:
Namespace (and pool) is inherited from snapshot if pool is not set explicitly, otherwise namespace is not inherited or set for a clone, only the pool attribute is set. Hence this is not compared in the common part of the if..else structure.